File: BLL/MapModels/CommunityModel.cs
using System.Collections.Generic;

namespace Business.MapModels
{
    /// <summary>
    /// Represents Community mapped model.
    /// </summary>
    public class CommunityModel : BaseModel
    {
        /// <summary>
        /// Community's title.
        /// </summary>
        public string Title { get; set; }

        /// <summary>
        /// Information about community.
        /// </summary>
        public string About { get; set; }

        /// <summary>
        /// Community's creator's id.
        /// </summary>
        public int CreatorId { get; set; }

        /// <summary>
        /// Community's creation date.
        /// </summary>
        public string CreationDate { get; set; }

        /// <summary>
        /// Community's members count.
        /// </summary>
        public int MembersCount { get; set; }

        /// <summary>
        /// Collection of community's topic models.
        /// </summary>
        public IEnumerable<TopicModel> PostModels { get; set; }

        /// <summary>
        /// Collection of community's rule models.
        /// </summary>
        public IEnumerable<RuleModel> RuleModels { get; set; }

        /// <summary>
        /// Collection of community's moderator models.
        /// </summary>
        public IEnumerable<UserModel> ModeratorModels { get; set; }
    }
}

File: Forum.WebApi/ForumAngularClient/src/app/models/community/community.model.ts
import {TopicModel} from "../topic/topic.model";
import {RuleModel} from "../rule/rule.model";
import {UserModel} from "../user/user.model";

export class CommunityModel {
  constructor(
    public id?: number,
    public title?: string,
    public about?: string,
    public creatorId?: number,
    public creationDate?: string,
    public membersCount?: number,
    public postModels?: TopicModel[],
    public ruleModels?: RuleModel[],
    public moderatorModels?: UserModel[]
  ) {
  }
}
